app/controllers/internal/configs_controller.rb

```diff
@@ -3,14 +3,21 @@ class Internal::ConfigsController < Internal::ApplicationController
 
   before_action :extra_authorization_and_confirmation, only: [:create]
 
+  SOCIAL_MEDIA_PLATFORMS = %i[twitter facebook github instagram twitch]
```

Comment:
I predefine the SOCIAL_MEDIA_PLATFORMS = %i[twitter facebook github instagram twitch youtube], and allow the user to enter a different username for each in the list.
I feel like this may be sufficient because there's only maybe 5 social media platforms, and we'd only show the platforms that the user fills in usernames for. It also puts a more control in our hands and leaves less unknowns and loopholes for breakage.